testsuite/MDAnalysisTests/lib/test_cutil.py

```diff
@@ -46,6 +46,43 @@ def test_unique_int_1d(values):
     assert res.dtype == ref.dtype
 
 
+@pytest.mark.parametrize('arrays', (
+    [],  # empty array
+    [1, 1, 1, 1, ],  # all identical
+    [2, 3, 5, 7, ],  # unique, sorted
+    [5, 2, 7, 3, ],  # unique, unsorted
+    [1, 2, 2, 4, 4, 6, ],  # duplicates, sorted
+    [1, 2, 2, 6, 4, 4, ],  # duplicates, unsorted
+))
+def test_inverse_unique_contiguous_1d_array(arrays):
+    # Generate unique, unsorted arrays.
+    array = np.array(arrays, dtype=np.intp)
+    unique_array = unique_int_1d_unsorted(array)
+
+    inverse = inverse_unique_contiguous_1d_array(array, unique_array)
+    ref = array
+    res = unique_array[inverse]
+    assert_equal(res, ref)
+    assert type(res) == type(ref)
+    assert res.dtype == ref.dtype
```

Comment:
Instead of relying on the logic of `unique_int_1d_unsorted`, could you please just test against an expected value? You can pass that in as another `parameterize` argument. That makes it far more of a [unit test](https://smartbear.com/learn/automated-testing/what-is-unit-testing/) and therefore more robust (it doesn't rely on `unique_int_1d_unsorted` working properly) and also more readable. It should be fairly doable for these short arrays.